[beta 15.6]Fixe le comptage des messages non lus dans les forums.

artragis opened the pull request:
| Q | R |
| --- | --- |
| Correction de bugs ? | [oui] |
| Nouvelle Fonctionnalité ? | [non] |
| Tickets (_issues_) concernés | #2873 |
# note pour la QA :

Il vous faudra 2 utilisateurs

avec l'utilisateur n°1
- créez un topic

avec l'utilisateur n°2
- répondez au topic
- vérifiez dans la liste "toutes les notifications" que le topic n'est pas en gras

avec l'utilisateur n°1
- vérifiez qu'il y a une notification, mais **avant** dans la lire, allez dans "toutes les notifications" et vérifiez que le topic est en gras.

artragis committed:
Fixe le comptage des messages non lus dans les forums.

pierre-24 commented:
Tu va rire, mais j'ai à chaque fois le comportement inverse.

Exemple: avant d'avoir cliqué sur la notif, c'est pas en gras

![screenshot from 2015-07-03 19 55 41](https://cloud.githubusercontent.com/assets/7889675/8503300/9859e4fe-21bd-11e5-81a7-cd0f8359f7a1.png)

Puis, après avoir cliqué, ça l'est : 

![screenshot from 2015-07-03 19 55 49](https://cloud.githubusercontent.com/assets/7889675/8503305/a3767c12-21bd-11e5-9ddd-b1a8914def6a.png)

artragis commented:
hum, c'est normal vu comment j'ai programmé ça. La chaleur, ça me réussit pas, vraiment.

pierre-24 commented:
On est deux, je te rassure.

Situphen commented:
Trois ?

artragis committed:
corrige le template 

pour que la logique ne soit plus inversée, on précalcule les topics qui sont suivi et pas déjà lu.

artragis commented:
Ca devrait être bon. Par contre, tu pourras vérifier la liste des topics au sein des forums pour être sûr? Parce que du coup je touche au template qui est utilisé partout.

artragis commented:
je me réponds à moi même : ça ne peut pas marcher : la logique inversée était là pour quelque chose. faut donc que je fasse un revert et que je change mon exclude.

artragis committed:
inverse l'exclusion pour que les "messages lus" soient ceux sélectionnés

artragis commented:
bon, j'attends la QA (le TU que j'ai fait _risque_ de foirer car je n'ai pas compter les topics qui ont été mis dans la fonction setup.
une fois que la Qa sera validé, je corrigerai le TU si besoin et je ferai un squash.

pierre-24 commented:
Pour la QA, on est bon, en tout cas, attendons donc Travis.

artragis commented:
tu es sûr que c'set bon côté QA? car là si le test foire comme il le fait ça voudrait dire qu'il y a encore des problèmes.

pierre-24 commented:
Ah, je te promet que c'était bien gras quand il fallait o_O

artragis commented:
ça l'est toujours?

Le 03/07/2015 20:50, Pierre Beaujean a écrit :

> Ah, je te promet que c'était bien gras quand il fallait o_O
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/2880#issuecomment-118408337.

pierre-24 commented:
Toujours OK :)

artragis commented:
je sèche là pour le test. je regarde demain matin quand la fatigue sera moindre

artragis commented:
passez moi un flingue.

pierre-24 commented:
Ok, mais un fusil à eau, alors (j'ai encore besoin de toi :p )

artragis commented:
yey !
Je fais le fameux squash et c'est mergeable (par contre j'ai encore touché au code, une n-ième vérif serait appréciée)

pierre-24 commented:
Ok aussi pour moi, let's go :)

SpaceFox commented:
Vu l'état de la prod, j'ai un gros doute sur le fonctionnement de cette PR. Je n'ai **plus un seul** sujet en gras dans les forums.

Situphen commented:
Je plussoie @SpaceFox ! Les sujets que je n'ai jamais visité ne sont pas en gras alors qu'ils devraient l'être.
